app/controllers/posts_controller.rb

```diff
@@ -5,5 +5,23 @@ def index
 
   def show
     @post = Post.find_by(id: params[:id].to_i)
+    @comment = Comment.new
+    @like = Like.new
+    @user = current_user
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = current_user
+    @post.save
+    redirect_to "/users/#{current_user.id}/posts/#{@post.id}"
+  end
```

Comment:
- [ ] in this method you should use the `respond_to` Rails helper method

Reply:
done! thanks alot